### DISCUSSION TITLE
Is Sentry SDK for PHP using a queue when sending data?

### DISCUSSION THREAD

**Discussion by ghost** [getsentry/sentry-laravel#691] [Q&A] [answered]
When en exception is captured, this package calls sentry/sdk wich queues the request and then the process continues leaving the sdk job to send the exception to sentry.io? Or it is a sync request to sentry.io? I saw promises are used.

**Comment by stayallive** [comment, marked as answer]
Hi @marius-ciclistu, short answer: currently everything is sent synchronously.

There is a exception, and that is for transactions from the performance part of the Sentry SDK, these are sent in a terminable middleware and are thus sent after the response is sent.

So the errors/exception are sent immediately (could have impact on response times), and performance tracing is sent after the response is sent to the client (so should not have any impact on response times).

There are some ways to improve response times and/or lower the impact of sending data:  https://docs.sentry.io/platforms/php/performance/#improve-response-time.

We used to have asynchronous sending (as far as that was possible in PHP) that's why the promises are still in the code.

**Reply by ghost** [reply]
Thank you. I also thought it could be handled in the termination middleware. See this https://stackoverflow.com/a/76195957/7309871

**Reply by stayallive** [reply]
Yes. However we don't feel comfortable to do this since when the application crashes because of memory limits, timeouts or other exceptions we might never send certain errors to Sentry. Feel free to implement this in your own error handler if you feel comfortable with that level of risk but for us the risk of not sending an error is to great to implement a system like that.

**Reply by ghost** [reply]
Then my first solution with dispatching jobs is better.  I did not thought of the memory errors. Thanks.

**Reply by stayallive** [reply]
Do keep in mind that if your queue crashes and/or is not running or you fail to write the job to the queue data can also be lost. That is why we don't have any native queue integration again because of the risk of dropping data. But again, considering those risks you can always implement them yourself if the mentioned Relay is not an option or the overhead is still unacceptable to you.

**Reply by ghost** [reply]
I know. It is a trade off. The error will be logged but not sent via mail. Logged in non sentry way.
